--- /glade/u/home/zarzycki/work/cesm-dev//components/cam/src/physics/clubb//advance_clubb_core_module.F90	2020-07-04 18:33:26.509549051 -0600
+++ /glade/u/home/zarzycki/TC_PBL/RCE.QPC6.ne0np4tcfplane.ne15x8.exp294.001/SourceMods/src.cam/advance_clubb_core_module.F90	2020-08-19 14:32:34.651827517 -0600
@@ -404,6 +404,12 @@
         irvm,          &
         irel_humidity, &
         iwpthlp_zt,    &
+!++CMZ
+        iinvrs_tau_bkgnd, &
+        iinvrs_tau_sfc,   &
+        iinvrs_tau_shear, &
+        ibrunt_vaisala_freq_sqd, &
+!--CMZ
         itau_no_N2_zm, &
         itau_xp2_zm,   &
         itau_wp2_zm,   &
@@ -754,6 +760,11 @@
        invrs_tau_wp3_zm,             & ! One divided by tau_wp3                       [s^-1]
        invrs_tau_N2_zm,              & ! One divided by tau with stability effects    [s^-1]
        invrs_tau_no_N2_zm,           & ! One divided by tau (without N2) on zm levels [s^-1] 
+!++CMZ
+       invrs_tau_bkgnd,              & ! One divided by tau_wp3 [s^-1]
+       invrs_tau_shear,              & ! One divided by tau with stability effects    [s^-1]
+       invrs_tau_sfc,                & ! One divided by tau (without N2) on zm levels [s^-1] 
+!--CMZ
        ustar,                        & ! Friction velocity  [m/s]
        tau_no_N2_zm,                 & ! Tau without Brunt Freq
        tau_wp2_zm,                   & ! Tau values used for advance_wp2_wpxp
@@ -1161,9 +1172,20 @@
 
         ustar = max( ( upwp_sfc**2 + vpwp_sfc**2 )**(one_fourth), ufmin )
 
-        invrs_tau_no_N2_zm = C_invrs_tau_bkgnd  / tau_const &
-         + C_invrs_tau_sfc * ( ustar / vonk ) / ( gr%zm - sfc_elevation + z_displace ) &
-         + C_invrs_tau_shear * zt2zm( zm2zt( sqrt( (ddzt( um ))**2 + (ddzt( vm ))**2 ) ) )
+!++CMZ
+        invrs_tau_bkgnd = C_invrs_tau_bkgnd / tau_const
+
+        invrs_tau_shear = C_invrs_tau_shear * zt2zm( zm2zt( sqrt( (ddzt( um ))**2 + (ddzt( vm ))**2 ) ) )
+
+        invrs_tau_sfc   =  C_invrs_tau_sfc * ( ustar / vonk ) / ( gr%zm - sfc_elevation + z_displace )
+                          !C_invrs_tau_sfc * ( wp2 / vonk /ustar ) / ( gr%zm -sfc_elevation + z_displace ) 
+
+        invrs_tau_no_N2_zm = invrs_tau_bkgnd + invrs_tau_sfc + invrs_tau_shear
+
+!        invrs_tau_no_N2_zm = C_invrs_tau_bkgnd  / tau_const &
+!         + C_invrs_tau_sfc * ( ustar / vonk ) / ( gr%zm - sfc_elevation + z_displace ) &
+!         + C_invrs_tau_shear * zt2zm( zm2zt( sqrt( (ddzt( um ))**2 + (ddzt( vm ))**2 ) ) )
+!--CMZ
 
 !        brunt_vaisala_freq_sqd_smth = zt2zm( zm2zt( brunt_vaisala_freq_sqd ) )
 !       The min function below smooths the slope discontinuity in brunt freq
@@ -1433,6 +1455,13 @@
       end if ! l_stability_correction
 
       if ( l_stats_samp ) then
+!++CMZ
+      call stat_update_var( iinvrs_tau_bkgnd,invrs_tau_bkgnd , stats_zm)
+      call stat_update_var( iinvrs_tau_sfc,invrs_tau_sfc , stats_zm)
+      call stat_update_var( iinvrs_tau_shear,invrs_tau_shear, stats_zm)
+      call stat_update_var( ibrunt_vaisala_freq_sqd,brunt_vaisala_freq_sqd_smth, stats_zm)
+      
+!--CMZ
       call stat_update_var( itau_no_N2_zm,tau_no_N2_zm , stats_zm)
       call stat_update_var( itau_xp2_zm,tau_xp2_zm , stats_zm)
       call stat_update_var( itau_wp2_zm,tau_wp2_zm , stats_zm)
