--- /glade/u/home/zarzycki/work/cesm-dev//components/cam/src/physics/clubb//stats_zm_module.F90	2020-07-04 18:33:26.556910474 -0600
+++ /glade/u/home/zarzycki/TC_PBL/RCE.QPC6.ne0np4tcfplane.ne15x8.exp295.001/SourceMods/src.cam/stats_zm_module.F90	2020-08-19 13:59:04.990971993 -0600
@@ -59,7 +59,13 @@
         ircp2,   &
         iSkw_zm, &
         iSkthl_zm, &
-        iSkrt_zm
+!++CMZ
+!        iSkrt_zm
+        iSkrt_zm,  &
+        iinvrs_tau_bkgnd, &
+        iinvrs_tau_sfc, &
+        iinvrs_tau_shear
+!--CMZ
 
     use stats_variables, only: &
         iupwp, & 
@@ -653,6 +659,32 @@
              l_silhs=.false., grid_kind=stats_zm )
         k = k + 1
 
+!++CMZ
+      case ('invrs_tau_bkgnd')
+        iinvrs_tau_bkgnd = k
+
+        call stat_assign( var_index=iinvrs_tau_bkgnd, var_name="invrs_tau_bkgnd", &
+             var_description="invrs tau of bkgnd on momentum levels [s-1]", var_units="s^-1", &
+             l_silhs=.false., grid_kind=stats_zm )
+        k = k + 1
+
+      case ('invrs_tau_sfc')
+        iinvrs_tau_sfc = k
+
+        call stat_assign( var_index=iinvrs_tau_sfc, var_name="invrs_tau_sfc", &
+             var_description="invrs tau of surface on momentum levels [s-1]", var_units="s^-1", &
+             l_silhs=.false., grid_kind=stats_zm )
+        k = k + 1
+
+      case ('invrs_tau_shear')
+        iinvrs_tau_shear = k
+
+        call stat_assign( var_index=iinvrs_tau_shear, var_name="invrs_tau_shear", &
+             var_description="invrs tau of shear on momentum levels [s-1]", var_units="s^-1", &
+             l_silhs=.false., grid_kind=stats_zm )
+        k = k + 1
+!--CMZ
+
       case ('tau_no_N2_zm')
         itau_no_N2_zm = k
 
@@ -2380,12 +2412,14 @@
                           var_units="-", l_silhs=.false., grid_kind=stats_zm )
         k = k + 1
 
-      case ( 'brunt_vaisala_freq_sqd' )
+!++CMZ (change var name from brunt_vaisala_freq_sqd to bv_freq_sqd)
+      case ( 'bv_freq_sqd' )
         ibrunt_vaisala_freq_sqd = k
-        call stat_assign( var_index=ibrunt_vaisala_freq_sqd, var_name="brunt_vaisala_freq_sqd", &
+        call stat_assign( var_index=ibrunt_vaisala_freq_sqd, var_name="bv_freq_sqd", &
              var_description="Brunt-Vaisala freqency squared, N^2 [1/s^2]", var_units="1/s^2", &
              l_silhs=.false., grid_kind=stats_zm )
         k = k + 1
+!--CMZ
 
       case ( 'Richardson_num' )
         iRichardson_num = k
