--- /glade/u/home/zarzycki/work/cesm-dev//cime/src/share/util//shr_flux_mod.F90	2020-07-04 18:32:37.173061594 -0600
+++ /glade/u/home/zarzycki/TC_PBL/RCE.QPC6.ne0np4tcfplane.ne15x8.exp297.001/SourceMods/src.share/shr_flux_mod.F90	2020-08-17 19:30:32.275651063 -0600
@@ -265,7 +265,16 @@
 
 
    qsat(Tk)   = 640380.0_R8 / exp(5107.4_R8/Tk)
-   cdn(Umps)  =   0.0027_R8 / Umps + 0.000142_R8 + 0.0000764_R8 * Umps
+!++CMZ
+! Large and Yeager 2009
+   cdn(Umps)  =  0.0027_R8 / min(33.0000_R8,Umps) + 0.000142_R8 + 0.0000764_R8 * min(33.0000_R8,Umps) - 3.14807e-13_r8 * min(33.0000_R8,Umps)**6
+! Capped Large and Pond by wind
+!   cdn(Umps)  =   0.0027_R8 / min(30.0_R8,Umps) + 0.000142_R8 + 0.0000764_R8 * min(30.0_R8,Umps) 
+! Capped Large and Pond by Cd
+!   cdn(Umps) = min(0.0025_R8, (0.0027_R8 / Umps + 0.000142_R8 + 0.0000764_R8 * Umps ))
+! Large and Pond
+!   cdn(Umps)  =   0.0027_R8 / Umps + 0.000142_R8 + 0.0000764_R8 * Umps
+!--CMZ
    psimhu(xd) = log((1.0_R8+xd*(2.0_R8+xd))*(1.0_R8+xd*xd)/8.0_R8) - 2.0_R8*atan(xd) + 1.571_R8
    psixhu(xd) = 2.0_R8 * log((1.0_R8 + xd*xd)/2.0_R8)
 
