--- /glade/u/home/zarzycki/work/cesm2_2_0//components/cam/src/physics/clubb//advance_xm_wpxp_module.F90	2020-10-19 15:40:52.659952957 -0600
+++ advance_xm_wpxp_module.F90	2021-01-28 14:58:45.289068501 -0700
@@ -102,6 +102,12 @@
         C7,  & 
         C7b,  & 
         C7c,  & 
+    !++CMZ
+    ! C10 and C13 are not used, so we'll leverage them as constants for splitting
+    ! C7 in the prog momentum time tendency equation (pressure term).
+        C10,  & 
+        C13,  & 
+    !--CMZ
         c_K6,  & 
         C6rt_Lscale0, &
         C6thl_Lscale0, &
@@ -333,7 +339,10 @@
     ! -------------------- Local Variables --------------------
 
     real( kind = core_rknd ), dimension(gr%nz) ::  & 
-      C6rt_Skw_fnc, C6thl_Skw_fnc, C7_Skw_fnc
+!++CMZ
+      !C6rt_Skw_fnc, C6thl_Skw_fnc, C7_Skw_fnc
+      C6rt_Skw_fnc, C6thl_Skw_fnc, C7_Skw_fnc, C7upwp, C13tmp
+!--CMZ
 
     ! Eddy Diffusion for wpthlp and wprtp.
     real( kind = core_rknd ), dimension(gr%nz) :: Kw6  ! wpxp eddy diff. [m^2/s]
@@ -478,6 +487,11 @@
                                      C7_Lscale0, wpxp_L_thresh, Lscale )
     end if ! l_use_C7_Richardson
 
+    !++ CMZ
+    C7upwp(1:gr%nz) = C10
+    C13tmp(1:gr%nz) = C13
+    !-- CMZ
+
     ! Damp C6 as a function of Lscale in stably stratified regions
     C6rt_Skw_fnc = damp_coefficient( C6rt, C6rt_Skw_fnc, &
                                      C6rt_Lscale0, wpxp_L_thresh, Lscale )
@@ -593,6 +607,9 @@
                                           up2, vp2,                                         & ! In
                                           low_lev_effect, high_lev_effect,                  & ! In
                                           C6rt_Skw_fnc, C6thl_Skw_fnc, C7_Skw_fnc,          & ! In
+                                          !++CMZ
+                                          C7upwp, C13tmp,                                   & ! In
+                                          !--CMZ
                                           lhs_diff_zm, lhs_diff_zt, lhs_ma_zt, lhs_ma_zm,   & ! In
                                           lhs_ta_wprtp,                                     & ! In
                                           rhs_ta_wprtp, rhs_ta_wpthlp, rhs_ta_wpup,         & ! In
@@ -2221,6 +2238,9 @@
                                             up2, vp2, &
                                             low_lev_effect, high_lev_effect, &
                                             C6rt_Skw_fnc, C6thl_Skw_fnc, C7_Skw_fnc, &
+                                            !++CMZ
+                                            C7upwp, C13tmp, &
+                                            !--CMZ
                                             lhs_diff_zm, lhs_diff_zt, lhs_ma_zt, lhs_ma_zm, &
                                             lhs_ta_wpxp, &
                                             rhs_ta_wprtp, rhs_ta_wpthlp, rhs_ta_wpup, &
@@ -2274,6 +2294,10 @@
       ivprcp,  &
       l_stats_samp
         
+    !++CMZ
+    use spmd_utils,      only: masterproc
+    !--CMZ
+        
     use parameters_model, only: & 
       sclr_dim, &  ! Variable(s)
       sclr_tol
@@ -2385,7 +2409,10 @@
       high_lev_effect   ! Index of the highest level that has an effect.
       
     real( kind = core_rknd ), dimension(gr%nz), intent(in) ::  & 
-      C6rt_Skw_fnc, C6thl_Skw_fnc, C7_Skw_fnc
+!++CMZ
+      !C6rt_Skw_fnc, C6thl_Skw_fnc, C7_Skw_fnc
+      C6rt_Skw_fnc, C6thl_Skw_fnc, C7_Skw_fnc, C7upwp, C13tmp
+!--CMZ
 
     integer, intent(in) :: &
       nrhs         ! Number of RHS vectors
@@ -2558,14 +2585,39 @@
        endif ! .not. l_implemented
 
        ! Add "extra term" and optional Coriolis term for <u'w'> and <v'w'>.
+       !++CMZ
+       !upwp_forcing = C7_Skw_fnc * wp2 * ddzt( um )
+       !vpwp_forcing = C7_Skw_fnc * wp2 * ddzt( vm )
+       if (masterproc) then
+         print *,'CMZ: C7upwp ',C7upwp(1),'      C7_Skw_fnc ',C7_Skw_fnc(1)
+       end if
+       if ( C7upwp(1) < 99.0_core_rknd ) then
+         upwp_forcing = C7upwp * wp2 * ddzt( um )
+         vpwp_forcing = C7upwp * wp2 * ddzt( vm )
+       else
        upwp_forcing = C7_Skw_fnc * wp2 * ddzt( um )
        vpwp_forcing = C7_Skw_fnc * wp2 * ddzt( vm )
+       endif
+       !--CMZ
 
        if ( l_stats_samp ) then
+       !++CMZ
+         if ( C7upwp(1) < 99.0_core_rknd ) then
+           call stat_update_var( iupwp_pr4, C7upwp * wp2 * ddzt( um ), &
+                                stats_zm )
+           call stat_update_var( ivpwp_pr4, C7upwp * wp2 * ddzt( vm ), &
+                                stats_zm )
+         else
           call stat_update_var( iupwp_pr4, C7_Skw_fnc * wp2 * ddzt( um ), &
                                 stats_zm )
           call stat_update_var( ivpwp_pr4, C7_Skw_fnc * wp2 * ddzt( vm ), &
                                 stats_zm )
+         endif
+          !call stat_update_var( iupwp_pr4, C7_Skw_fnc * wp2 * ddzt( um ), &
+          !                      stats_zm )
+          !call stat_update_var( ivpwp_pr4, C7_Skw_fnc * wp2 * ddzt( vm ), &
+          !                      stats_zm )
+        !--CMZ
        endif ! l_stats_samp
 
        call diagnose_upxp( upwp, thlm, wpthlp, um, &               ! Intent(in)
